app.rb

```diff
@@ -0,0 +1,146 @@
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'books'
+require_relative 'rentals'
+
+class App
+  private
+
+  def create_student(arr_person)
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permision? [Y/N]: '
+    get_permision = gets.chomp.upcase
+    until %w[Y N].include?(get_permision)
+      puts 'Please Enter valid permision'
+      get_permision = gets.chomp.upcase
+    end
+    case get_permision
+    when 'Y'
+      permision = true
+    when 'N'
+      permision = false
+    end
+    new_student = Student.new(age, name, permision)
+    arr_person.push(Student.new(age, name, permision)) unless arr_person.include?(Student.new(age, name, permision))
+    print "\n Student with Name: ", new_student.name, ' Permision: ', new_student.can_use_services, ' Age: ',
+          new_student.age, " created successfully\n\n"
+  end
+
+  def create_teacher(arr_person)
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp.downcase
+    new_teacher = Teacher.new(age, specialization, name)
+    unless arr_person.include?(Teacher.new(age, specialization,
+                                           name))
+      arr_person.push(Teacher.new(age, specialization,
+                                  name))
```

Comment:
- [ ] Similarly, your app is breaking in an attempt to create a person of type teacher. Kindly ensure to improve upon this aspect.

![image](https://user-images.githubusercontent.com/12745474/211175672-a35e68ad-95e4-4829-b51c-c45d2b7b1425.png)